breast_basal/scripts/11_mutation_subset.R

```diff
@@ -0,0 +1,42 @@
+## This script subsets the patient barcodes with mutations from the entire methylation dataset
+
+## Load libraries-----------------
+library(tidyverse)
+library(dplyr)
+library(stringr)
+
+
+## Load data-----------
+Maf_Basal <- read.csv("breast_basal/data/mutations.csv")
+sample_annotation <- get(load("breast_basal/data/BRCA_Basal_sample_annotations.rda"))
+
+## DMA----------
+Maf_Basal_filtered <- Maf_Basal |> 
+  mutate(VAF = t_alt_count/t_depth) %>% 
+  filter(VAF > 0.05, 
+         t_alt_count >= 3,
+         t_depth > 30,
+         n_depth >10)
+
+## Separate the samples into cancer and normal
+cancer_barcodes <- sample_annotation  |> 
+  filter(sample.type == "Cancer")  |> 
+  select(primary)
+normal_barcodes <- sample_annotation  |> 
+  filter(sample.type == "Normal")  |> 
+  select(primary)
+
+## Match methylation patient barcodes with mutations for the cancer samples
+cancer_mutations_subset <-  Maf_Basal_filtered %>% 
+  filter((substr(x = Tumor_Sample_Barcode, start = 1, stop = 15)) %in% cancer_barcodes$primary)
+
+## Match methylation patient barcodes with mutations for the normal samples
+normal_mutations_subset <- Maf_Basal_filtered %>% 
+  filter((substr(x = Matched_Norm_Sample_Barcode, start = 1, stop = 15)) %in% normal_barcodes$primary)
```

Comment:
From our understanding, yes, the MAF file only contains barcodes for cancer patients. We subset the mutations based on the "sample_annotations" file, which contains annotations for both cancer and normal samples. The "normal_mutations_subset" refers to those barcodes that have the same "Matched_Norm_Sample_Barcode" as the "Normal" samples in the "sample_annotations" file - they don't actually refer to normal patients. Maybe the naming scheme we used is a bit ambiguous.

Reply:
ok - I'm not sure I understand why we are doing this

What I mean is, if the MAF file contains mutations from cancer samples only, its TCGA barcodes should never match the set of normal samples. Besides, I think we only want to consider samples from cancer patients in terms of mutation. Do we ever find cases in which we incorporate mutations from normal sample in the analysis?

Reply:
Yes, you are right - the MAF file contains mutations from cancer patients only. Subsetting the MAF file using the tissue type ("primary") from "sample_annotations" would have been sufficient, instead of dividing them based on the tissue type first and then subsetting. As a result, "normal_mutations_subset" (which is a subset of "cancer_mutations_subset" in this case) has been included twice in "all_mutations_subset". We apologize for this oversight.

What this means is "all_mutations_subset" that is used for downstream analysis needs to be changed. What effect this has on downstream analysis - we don't know yet. We have already made this change on shared_projects in the server and are currently performing the downstream analysis again. We will notify you about the results.

Reply:
Yes, we do incorporate mutations from normal samples in the analysis since the MAF file contains some normal tissue samples from cancer patients, which are included in the 'sample annotation' file. We wanted to preserve all the samples present in the 'sample annotation' file; therefore, we included the normal samples. If we didn't include all the samples from 'sample annotation' file, the datasets used for GMA and DMA would not be the same, so we wouldn't be able to compare their results. If you think this is not the right methodology, we can change it.